### PR TITLE
Add .formatter.exs and format everything

### PR DESCRIPTION
**Pull request by devonestes**
Before I update this for memory measurements, I'm just getting the
formatting and updating stuff out of the way.

Luckily this isn't so bad. After nearly a year of using the formatter,
the stuff that isn't formatted looks really weird to me now, actually!

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-elixir 1.5.2
-erlang 20.0
+elixir 1.7.3-otp-21
```
Comment (PragTob) [inline .tool-versions]: the elixir version looks off - shouldn't it just be 1.7.3?

**Comment by PragTob** [comment]
ah just small tihing about the .tool-versions looking odd, otherwise good of course

**Comment by PragTob** [comment]
oh and shouldn't we add it to travis?

File: .tool-versions (hunk 1 above)
Comment (devonestes) [reply .tool-versions]: You can now specify an Elixir Version that was compiled with a certain OTP version in this way. I find doing this makes Dialyzer much happier. It’s a pretty sweet asdf feature.

Comment (PragTob) [reply .tool-versions]: Interesting, thanks.

Not just dialyzer but some other dependency compilation things I imagine...

**Pull request merged**